integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -1511,7 +1514,16 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
         }
         DataTypeInfo("decimal", precision, scale)
       case _ =>
-        throw new MalformedCarbonCommandException("Data type provided is invalid.")
+        if (isColumnRename) {
+          dataType match {
```

Comment:
why here match `dataType` again which is already match in line 1496. Seems this func need to be refactored

Reply:
ok, refactored, if rename operation then just return the datatypeInfo object, else throw invalid datatype exception as before